statefun-e2e-tests/statefun-smoke-e2e-java/pom.xml

```diff
@@ -27,68 +27,40 @@ under the License.
 
     <artifactId>statefun-smoke-e2e-java</artifactId>
 
-    <properties>
-        <commons-math3.version>3.5</commons-math3.version>
-        <additional-sources.dir>target/additional-sources</additional-sources.dir>
-    </properties>
-
     <dependencies>
-        <!-- Stateful Functions -->
+        <!-- Remote Java function dependencies -->
         <dependency>
             <groupId>org.apache.flink</groupId>
             <artifactId>statefun-sdk-java</artifactId>
             <version>${project.version}</version>
         </dependency>
+        <dependency>
+            <groupId>io.undertow</groupId>
+            <artifactId>undertow-core</artifactId>
+            <version>1.4.18.Final</version>
+        </dependency>
+        <dependency>
+            <groupId>com.google.protobuf</groupId>
+            <artifactId>protobuf-java</artifactId>
+            <version>${protobuf.version}</version>
+        </dependency>
```

Comment:
May I know the rationale behind adding this? 
The protobuf-java was in the dependency tree brought in by smoke-e2e-driver. 
If we explicitly specify the dependency here, that shall mean it's required by the smoke-e2e-java.  However I do not see the usage.

Reply:
Explained in https://github.com/apache/flink-statefun/pull/248/files#r684575758.